src/Geolocation/SettingsGeolocation.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * Geolocation Settings class.
+ *
+ * @package EightshiftForms\Geolocation
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Geolocation;
+
+use EightshiftForms\Settings\Settings\SettingsDataInterface;
+use EightshiftForms\Settings\SettingsHelper;
+use EightshiftFormsVendor\EightshiftLibs\Services\ServiceInterface;
+
+/**
+ * SettingsGeolocation class.
+ */
+class SettingsGeolocation implements SettingsDataInterface, ServiceInterface
+{
+	/**
+	 * Use general helper trait.
+	 */
+	use SettingsHelper;
+
+	/**
+	 * Filter settings sidebar key.
+	 */
+	public const FILTER_SETTINGS_SIDEBAR_NAME = 'es_forms_settings_sidebar_geolocation';
+
+	/**
+	 * Filter settings key.
+	 */
+	public const FILTER_SETTINGS_GLOBAL_NAME = 'es_forms_settings_global_geolocation';
+
+	/**
+	 * Filter settings global is Valid key.
+	 */
+	public const FILTER_SETTINGS_GLOBAL_IS_VALID_NAME = 'es_forms_settings_global_is_valid_geolocation';
+
+	/**
+	 * Settings key.
+	 */
+	public const SETTINGS_TYPE_KEY = 'geolocation';
+
+	/**
+	 * Geolocation Use key.
+	 */
+	public const SETTINGS_GEOLOCATION_USE_KEY = 'geolocation-use';
+
+	/**
+	 * Register all the hooks
+	 *
+	 * @return void
+	 */
+	public function register(): void
+	{
+		\add_filter(self::FILTER_SETTINGS_SIDEBAR_NAME, [$this, 'getSettingsSidebar']);
+		\add_filter(self::FILTER_SETTINGS_GLOBAL_NAME, [$this, 'getSettingsGlobalData']);
+		\add_filter(self::FILTER_SETTINGS_GLOBAL_IS_VALID_NAME, [$this, 'isSettingsGlobalValid']);
+	}
+
+	/**
+	 * Determine if settings global are valid.
+	 *
+	 * @return boolean
+	 */
+	public function isSettingsGlobalValid(): bool
+	{
+		$isUsed = $this->isCheckboxOptionChecked(self::SETTINGS_GEOLOCATION_USE_KEY, self::SETTINGS_GEOLOCATION_USE_KEY);
+
+		if (!$isUsed) {
+			return false;
+		}
+
+		return true;
```

Comment:
No need for this tbh, the `isCheckboxOptionChecked` method uses `in_array` which will return a boolean, so just return that instead (without the variable).